src/Marks.tsx

```diff
@@ -249,8 +256,28 @@ interface EventTooltipProps {
 
 const EventTooltip = ({ type, y, parentWidth, text, triggerRef }: EventTooltipProps) => {
   const classes = useTooltipStyle()
-  const width = type === 'period' ? 180 : 100
+
+  const textLines = text.split('\n')
+  const numLinesInText = textLines.length
+  const isMultiLineText = numLinesInText > 1
+  const tooltipTextPadding = 15
+  let width = 0
```

Comment:
Extract `width` calculation into function in order to avoid mutability (i.e. replace `let` with `const`)

Reply:
Done, I have extracted the calculation of Tooltip dimensions into a separate function. While I was at it, I also created a small FC for rendering the Tooltip text itself. Also, I have now made the tooltip text render to use `<tspan>s` wrapped inside a `<text>` element for both multi-line and 1-line texts. This way the rendering logic is much simplified.